colcon_core/task/python/test/pytest.py

```diff
@@ -71,23 +71,32 @@ def match(self, context, env, setup_py_data):  # noqa: D102
         env = dict(env)
 
         if has_test_dependency(setup_py_data, 'pytest-cov'):
-            args += [
-                '--cov=' + str(PurePosixPath(
-                    *(Path(context.args.path).parts))),
-                '--cov-report=html:' + str(PurePosixPath(
-                    *(Path(context.args.build_base).parts)) / 'coverage.html'),
-                '--cov-report=xml:' + str(PurePosixPath(
-                    *(Path(context.args.build_base).parts)) / 'coverage.xml'),
-            ]
-            # use --cov-branch option only when available
-            # https://github.com/pytest-dev/pytest-cov/blob/v2.5.0/CHANGELOG.rst
-            from pytest_cov import __version__ as pytest_cov_version
-            if parse_version(pytest_cov_version) >= parse_version('2.5.0'):
+            try:
+                from pytest_cov import __version__ as pytest_cov_version
+            except ImportError:
+                logger.warning(
+                    'Test coverage will not be produced for package '
+                    "'{context.pkg.name}' since the pytest extension 'cov' "
+                    'was not found'.format_map(locals()))
```

Comment:
With / without the patch if the version of `cov` isn't new enough it is skipping the coverage report silently. So warning in the case where `cov` is not available seems inconsistent.